google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -1061,14 +1074,25 @@ def files prefix: nil, delimiter: nil, token: nil, max: nil,
         #   file = bucket.file "path/to/my-file.ext"
         #   puts file.name
         #
-        def file path, generation: nil, skip_lookup: nil, encryption_key: nil
+        def file path,
+                 generation: nil,
+                 if_generation_match: nil,
```

Comment:
Do these parameters get propagated across all Object operations that support these preconditions?
Similar question for bucket.

Reply:
No. Is that the desired behavior? To store the precondition values in the `File` instance and propagate them to `rewrite`, `update`, etc? It sounds a bit tricky and would require additional docs to clarify the behind-the-scenes behavior, but I can attempt it.